app/components/markdown/markdown_link/markdown_link.js

```diff
@@ -59,13 +58,10 @@ export default class MarkdownLink extends PureComponent {
         if (match) {
             if (match.type === DeepLinkTypes.CHANNEL) {
                 const {intl} = this.context;
-                this.props.actions.handleSelectChannelByName(match.channelName, match.teamName, errorBadChannel.bind(null, intl), intl);
+                this.props.actions.handleSelectChannelByName(match.channelName, match.teamName, errorBadChannel, intl);
```

Comment:
handleSelectChannelByName can be destructured from line 44 out of `actions`.   This will avoid long chains of code.